### PR TITLE
Keep the region a censored index leaves for the slope it shares

### PR DESCRIPTION
**Pull request by choxos**
> Stacked on #78, which supplies the exact-sum primitive this generalizes. Base retargets to `main` when #78 merges.

## What was wrong

An index whose rows are all censored pins nothing, so it removes no power of the auxiliary's width and its `aux_order` is `0`. The guard read that as "the slope is free". It is not the same thing. Those rows confine the coefficients to a **region**:

```
lower_i <= mu_index + beta' x_i <= upper_i
```

and under `model = "spfa"` with `aux_by = "none"` the comparator shares that slope **and** that auxiliary, so its ridge has to sit inside the region.

Take an index left-censored at `t = 1` on `x = 0` beside one right-censored at `t = 4` on `x = 1`. In the zero-scale limit it needs `mu_index <= 0` and `mu_index + beta >= log 4`, so `beta >= log 4`. A binary comparator with exact event times `(1, 1, 2)` has integration atoms 0 and 1 and two distinct targets, so its exact fit needs `beta = +/- log 2`. Neither is reachable. Every path to the boundary leaves one side with a vanishing probability, which beats the other's polynomial growth, and the posterior is proper.

Measured on that data, profiling over `(mu_index, mu_comparator, beta)`:

| `s` | 0.2 | 0.15 | 0.1 | 0.07 | 0.05 |
|---|---|---|---|---|---|
| `d log L / d log s` | +2.49 | +3.76 | +7.86 | +18.68 | +38.75 |

A growing positive slope is `exp(-c / s^2)`, not a power. The guard refused it as improper.

## What changed

The region travels to the comparator check as a new `index_region` attribute. There, `mu_index` is eliminated between a row with a finite lower end and one with a finite upper end (Fourier-Motzkin), leaving one condition on the slope per pair:

```
lower_i - upper_j <= beta (x_i - x_j)
```

Every candidate slope `(u[2] - u[1]) / (z_b - z_a)` is tested against it. The comparison is **cross-multiplied rather than divided** (a floating-point solve certifies nothing) and its sign is taken exactly, by a new `.exact_sum_sign()` generalizing #78's zero test. Where any of `C`, `D`, `p` or `q` did not survive its own subtraction, the answer is undecided rather than guessed.

**The verdict is three-valued, and it has to be:**

| state | meaning | action |
|---|---|---|
| strictly inside | the divergent slope is admitted | refuse, as before |
| strictly outside | no slope the index leaves carries this divergence | pass the arm over |
| on the boundary | `mu_index` is pinned to a point, costing a power not counted here | report |

The boundary case is why this is not a two-way test. There the index's slice in `mu_index` is a single point rather than an interval, so the true net rate is `worst - 1`; a silent pass would hide an improper fit at `worst >= 2`, and a refusal would be wrong at `worst = 1`. It is reported.

The filter also reaches `.grid_hits_targets()` through a new `admits` argument, so past the grid's reach only an allocation whose slope the index admits can certify.

## Acceptance

All five probes from the review, run natively through the public path with the fitting backend replaced by a sentinel:

| case | expected | before | now |
|---|---|---|---|
| E01, index needs `beta >= log 4` (proper) | reach backend | **refused as improper** | reach backend |
| E01 control, `beta >= log 0.5` (improper) | refuse | refuse | refuse |
| E02, exact Gaussian grid (improper) | refuse | reached backend | refuse |
| prior uniform grid, 8 points | refuse | refuse | refuse |
| prior uniform grid, 256 points | refuse | refuse | refuse |

## Evidence

- Three states reachable and correct at the switchover: `right_time = 4` outside, `= 2` exactly on the boundary, `= 1` and `= 0.5` inside. The improper control's profile grows as `s^-3` (measured -3.0000 per decade) and is still refused.
- **Centering invariance**: the elimination uses only covariate *differences*, so centering the design leaves every verdict unchanged. Tested at centers 0, 0.5 and 1e6.
- **Gating**: under `relaxed` the comparator has its own `beta_comparator`, and under `aux_by = ".study"` its scale reaches the boundary independently, so the region restricts nothing in either case. Both still refuse the same arm.
- **Failing-first**: every new assertion run against the base commit fails there.
- Pure-R suite: `FAIL 0 | WARN 42 | SKIP 93 | PASS 5083` (5036 at the base). Lints at baseline, `codetools::checkUsagePackage` clean, no duplicate top-level definitions.
- Cost, on a grid where no pair is ever admitted so the scan cannot exit early: 0.02 s at 256 nodes, 0.18 s at 1024, 2.5 s at 4096. The common case exits at the first admitted pair.

## Scope

One declared covariate, where the region's projection onto the slope is an interval. Wider designs leave a polyhedron whose projection is not read off pairs; those are left unrestricted exactly as before, and the limit is stated in the guard's documented scope alongside the enumeration's.

### CODE REVIEW

File: R/mlumr.R [hunk 1]
```diff
@@ -1184,6 +1379,24 @@
...
+  # its own `beta_comparator` and the region says nothing about it; under
+  # `aux_by = ".study"` the comparator's scale reaches its boundary on its
+  # own, with the index's at whatever value it likes, so the region is
+  # satisfied everywhere and says nothing there either.
+  admits <- if (shared_aux && identical(model, "spfa") &&
+                  !is.null(index_region)) {
+    function(tg) {
+      u <- sort(unique(as.numeric(tg)))
+      if (length(u) < 2L) return(NULL)
+      .index_slope_admits(index_region, u[2L] - u[1L])
```
[the remark this answers is not in the document]
Comment (choxos) [reply R/mlumr.R]: Already addressed in 4d8aab1, before this finding landed; the head it was written against is behind. `.index_slope_admits()` now takes the two targets rather than their difference, computes `p_err` itself, and folds it into the bound that every verdict has to dominate:

```r
> names(formals(mlumr:::.index_slope_admits))
[1] "region" "u1" "u2"
> edge <- mlumr:::.index_slope_admits(
+   list(X = cbind(1, c(0,1)), lower = c(-Inf, 1), upper = c(0, Inf)), 1e-20, 1)
> edge(0, 1)
[1] NA
```

That is the case you describe: `1 - 1e-20` rounds to `1`, the rounded numerator reads the region as an equality while the true slope violates it, and nothing is asserted.

The reason it is a bound rather than a rejection is that rejecting every inexact operand was measured to be far too strict: none of the four operands survives its own subtraction on ordinary data, and requiring all four to be exact left 2045 of 3266 generated regions whose slope really was admitted unsettled, which downgrades correct refusals to warnings. So the errors are carried as values, the perturbation is bounded above, the cheap four-term value bounded below, and a verdict stands only where the second dominates the first. Checked against exact rational arithmetic on those 3266 cases: no contradiction in either direction, and exactly one case left unsettled, the one the exact arithmetic also calls a boundary. Regression test `a rounded operand is bounded, not discarded`.

File: R/mlumr.R [hunk 2]
```diff
@@ -530,6 +704,15 @@
...
     # decided.
     pair <- is.finite(dz) & dz != 0
+    # A pair whose slope the index does not admit is not a candidate at all,
+    # however exactly it carries the targets, since the ridge it would form
+    # sits where the index's own probability goes to zero. An undecided one
+    # cannot certify and cannot be excluded either.
+    if (!is.null(admits)) {
+      ad <- admits(z0, z)
+      if (any(pair & (is.na(ad) | ad == 0))) open_slope <- TRUE
+      pair <- pair & !is.na(ad) & ad > 0
```
[the remark this answers is not in the document]
Comment (choxos) [reply R/mlumr.R]: Confirmed and fixed in 10eee2a. Reproduced with your own example first:

```r
> reg <- list(X = cbind(1, c(0,1)), lower = c(-Inf, 1), upper = c(0, Inf))
> ad  <- mlumr:::.index_slope_admits(reg, 0, 1)
> ad(0, c(0, 1, 2))
[1] -1  0 -1                      # the slope-1 pair is on the boundary
> mlumr:::.grid_hits_targets(matrix(c(0,1,2), ncol=1L), c(0,1,3), admits = ad)
[1] NA
attr(,"declined")
[1] "slope"                        # warned, though the pair misses the target at 3
```

Taken as written: a pair admitted only on the boundary, or not settled, now runs through the same target tests as an admitted one while never being able to certify, and the boundary is reported only where such a pair is still alive after every target determinant. The same call now returns `FALSE` with no `declined` attribute, while a boundary pair that DOES carry every target (add the node at 3, or use targets `(0, log 2, log 4)` on nodes `(0, 1, 2)`) still reports. Both directions are regression tests now.

**Comment by choxos** [comment]
@codex review

File: R/mlumr.R [hunk 3]
```diff
@@ -1235,8 +1505,35 @@
...
+      # Outside is a certificate that this arm's divergence cannot happen at
+      # any slope the index leaves: every path to the boundary leaves the
+      # index with a vanishing probability, which beats a polynomial. On the
+      # boundary the region's slice in `mu_index` is a single point rather
+      # than an interval, so the index costs a power that is not counted
+      # here; that is reported rather than refused or passed over.
+      if (identical(slope_state, "outside")) next
+      if (identical(slope_state, "boundary")) {
+        slope_open <- TRUE
+        next
```
[the remark this answers is not in the document]
Comment (choxos) [reply R/mlumr.R]: Correct, and a gap 10eee2a left in the caller. Reproduced with your exact numbers, built through `add_integration()` so the nodes really are `(0, 1, 2)`:

```r
> z
[1] 0 1 2
> mlumr:::.admitted_slope_state(matrix(z, ncol = 1L), ad)
[1] "boundary"                    # the scan, on its own
> mlumr:::.grid_hits_targets(matrix(z, ncol = 1L), c(0, 1, 3), admits = ad)
[1] FALSE                          # the enumeration, which rules the pair out
```

The caller took the first and never reached the second.

Fixed in ce40501 by putting the two in the order you name: past the reach the enumeration decides, and the standalone scan is kept only for arms within the reach, where any two distinct nodes carry the two targets and there is no further target to survive, so the scan IS the whole question there. The same case now leaves the caller silent (`expect_silent` plus `expect_false` in the new regression test `past the reach the enumeration decides the slope, not the scan`, which asserts the scan-versus-enumeration disagreement explicitly so the ordering cannot regress unnoticed).

Suite `FAIL 0 | PASS 5108`, all five acceptance cases still correct.

**Comment by choxos** [comment]
Closed by GitHub when its base branch was deleted on the merge of #78, not by intent. Continued as #81 against `main`: same tree, verified byte-identical to this branch's head, with the three findings answered here carried over.
